src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -8845,6 +8855,104 @@ public void Version()
             }
         }
 
+        [Fact]
+        public void RefOut()
+        {
+            var dir = Temp.CreateDirectory();
+            dir.CreateDirectory("ref");
+
+            var src = dir.CreateFile("a.cs");
+            src.WriteAllText(@"class C { public static void Main() { } }");
+
+            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
+            var csc = new MockCSharpCompiler(null, dir.Path, new[] { "/nologo", "/out:a.dll", "/refout:ref/a.dll", "/deterministic", "a.cs" });
+
+            int exitCode = csc.Run(outWriter);
+            Assert.Equal(0, exitCode);
+
+            var refDll = Path.Combine(dir.Path, "ref\\a.dll");
+            Assert.True(File.Exists(refDll));
+            var refPeStream = File.OpenRead(refDll);
+
+            using (var refPeReader = new PEReader(refPeStream))
+            {
+                var metadataReader = refPeReader.GetMetadataReader();
+
+                // The types and members that are included needs further refinement.
+                // See issue https://github.com/dotnet/roslyn/issues/17612
+                AssertEx.SetEqual(metadataReader.TypeDefinitions.Select(t => metadataReader.Dump(t)),
+                    new[] { "TypeDef:<Module>", "TypeDef:C" });
+
+                AssertEx.SetEqual(metadataReader.MethodDefinitions.Select(t => metadataReader.Dump(t)),
+                    new[] { "MethodDef: Void Main()", "MethodDef: Void .ctor()" });
+
+                // ReferenceAssemblyAttribute is missing.
+                // See issue https://github.com/dotnet/roslyn/issues/17612
+                AssertEx.SetEqual(
+                    metadataReader.CustomAttributes.Select(a => metadataReader.GetCustomAttribute(a).Constructor)
+                        .Select(c => metadataReader.GetMemberReference((MemberReferenceHandle)c).Parent)
+                        .Select(p => metadataReader.GetTypeReference((TypeReferenceHandle)p).Name)
+                        .Select(n => metadataReader.GetString(n)),
+                    new[] { "CompilationRelaxationsAttribute", "RuntimeCompatibilityAttribute", "DebuggableAttribute" });
+
+                // no method implementations
+                foreach (var typeDef in metadataReader.TypeDefinitions)
+                {
+                    Assert.Equal(0, metadataReader.GetTypeDefinition(typeDef).GetMethodImplementations().Count());
```

Comment:
>Assert.Equal(0, metadataReader.GetTypeDefinition(typeDef).GetMethodImplementations().Count()); [](start = 20, length = 94)

This doesn't actually verify that there were no method bodies emitted, which I assume is the intent here.